eventsourcing/src/main/java/org/axonframework/eventsourcing/MultiStreamableMessageSource.java

```diff
@@ -1,84 +0,0 @@
-/*
- * Copyright (c) 2010-2023. Axon Framework
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.axonframework.eventsourcing;
-
-import org.axonframework.messaging.StreamableMessageSource;
-
-/**
- * Implementation which allows for tracking processors to process messages from an arbitrary number of sources. The
- * order in which messages from each stream are consumed is configurable but defaults to the oldest message available
- * (using the event's timestamp). When the stream is polled for a specified duration, each stream is called with
- * {@code MultiSourceBlockingStream#hasNextAvailable()} except for the last stream configured by the
- * {@link MultiStreamableMessageSource.Builder#addMessageSource(String, StreamableMessageSource)} or by explicit
- * configuration using {@link MultiStreamableMessageSource.Builder#longPollingSource(String)}. This stream long polls
- * for a fraction of the specified duration before looping through the sources again repeating until the duration has
- * been met. This ensures the highest chance of a consumable message being found.
- *
- * @author Greg Woods
- * @since 4.2
- * @deprecated In favor of {@link org.axonframework.eventhandling.MultiStreamableMessageSource}. This class belongs in
- * the {@code messaging} module instead of the {@code eventsourcing} module.
```

Comment:
This might be an exception to the rule and something we can easily add a rewrite rule for, WDYT? Although it should be easily fixed by an import suggestion.

Reply:
Agreed, adding this as a rule is simple enough. I've added it to the `api-changes.md` so we can transform that into rewrite rules at a later stage!